primitives/src/lib.rs

```diff
@@ -79,6 +80,94 @@ pub struct ExchangeIntention<AccountId, Balance, IntentionID> {
 	pub intention_id: IntentionID,
 }
 
+#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, PartialOrd, Ord, MaxEncodedLen, RuntimeDebug, TypeInfo)]
+#[repr(u8)]
+pub enum ReserveIdentifier {
+	Nft,
+	Marketplace,
+
+	// always the last, indicate number of variants
+	Count,
+}
+
+pub mod nft {
+	use super::*;
+
+	#[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub enum ClassType {
+		Bare = 0_isize,
+		Marketplace = 1_isize,
```

Comment:
I wanted to have a default fallback category in case marketplace nfts will behave differently (e.g. some permission restrictions) in future. Marketplace can be the default type and can be named also differently. Any suggestions?

Reply:
it will be really hard to migrate between classes, so most likely you will have to enable this functionality for more classes. so marketplace wont be most probably 1:1 but 1:n

Reply:
therefore we don't really need this class to exist in my opinion, it is just misleading